### ISSUE TITLE
Controllers not working

### ISSUE THREAD

**Issue by ChristianBliss** [elishacloud/Silent-Hill-2-Enhancements#298]
Hi!
I've run into trouble trying to play with controllers. I've tried many times and checked everything you guys pointed out in the troubleshooting site on your website but I can't get to work any controller. I've tried xbox 360 and ps4 following the instructions carefully. Funny thing is that I remember playing the vanilla version of the game with the ps4 controller just fine...

I'd like to point out that I have installed vJoy driver (use it for midi purposes), but I tried to disable it and didn't seem to affect anything in game.

I'll leave the screenshot of the game's directory and my devices pane, as well as the 
[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/4917773/d3d8.log) file, thank you for your time!
![directorySnap](https://user-images.githubusercontent.com/26469715/87406727-6a3d8c80-c5c1-11ea-858a-126ee291559d.PNG)
![devices](https://user-images.githubusercontent.com/26469715/87407609-b3421080-c5c2-11ea-9ded-ae27dfb25a74.PNG)

**Comment by Badore90** [comment]
I use a wireless X360 pad with one of those chinese usb receivers, never had a problem. I also used a Ps2 gamepad with no issues.

I see you're missing the Xinput Plus files. They are for the vibration, the pad should work without them too (the ones I had at least). To try is cheap anyway.

**Comment by Polymega** [comment]
Silent Hill 2 PC detects and uses the first gamepad you connect to your computer. So if you have any other devices plugged into your PC that your computer detects/thinks is a gamepad (even though it isn't), it won't use the actual gamepad you're wanting to use. (This is the most common reason why the game does not detect the gamepad.)

This sounds silly, I know, but trying temporarily disconnecting things like your Razer mouse. I know Razer is a gaming-focused PC peripheral brand, so _maybe_ their mouse is being detected as a gamepad on your PC.

Basically, we want to use a process of elimination to determine what is causing the game to not detect your intended gamepad-of-choice. My gut says there's another device already connected to your PC as a gamepad, so therefor SH2 PC wants to use that device instead.

PS - I've asked team member Silent if we can possibly add a fix to allow _any and all_ gamepad inputs to be accepted by SH2 PC at all times, so we'd no longer have to worry about such an old, arcane restriction.

**Comment by Polymega** [comment]
Also, have you tried using a controller directly plugged into you PC? As in, **not** wireless, just to see if that works?

**Comment by ChristianBliss** [comment]
First of all, thank you both for your reply!!

1. I tried using xinput plus but didn't make any difference.

2. The gaming mouse was a good call actually, but unfortunately isn't the solution. I tried disconnecting both the mouse and keyboard with no results. I believe the same as you, there must be something in the background acting as a device... 
And yes, I tried wired controllers, actually that wireless controller from the screenshot is the ps4 dualshock plugged in via usb cable. Used both dualshock 4 and xbox 360 wired controller.

Oh, and that's amazing that you asked for a fix like that!! I hope it comes through and happens eventually. In the meantime I'll keep looking into it, see if there's any software or something interfering with devices in the background.

I'll keep this thread posted if anything comes up.

**Comment by Polymega** [comment]
Sorry I can't think of any better, concrete solutions off the top of my head. I remember, as another example, that someone bought a USB adapter to use [their Wii sensor bar ](https://www.amazon.com/Dolphin-Infrared-Nintendo-Console-Compatible/dp/B083GCFDBK/ref=sr_1_8?dchild=1&keywords=wii+sensor+bar+usb&qid=1594758009&sr=8-8)on their PC. It turned out for him that his PC detected that USB device as a gamepad, so he had to disconnect it in order to use his controller with SH2 PC.

**Comment by ChristianBliss** [comment]
No worries! I really appreciate your suggestions and help overall!

Tomorrow I'll check again, and will let you know if I find a solution.

**Comment by elishacloud** [comment]
@ChristianBliss, try with this build here: [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/4922188/d3d8.zip)

I added some logging so that you can see what controllers are available on your system and which one Silent Hill 2 is using.  this may help you troubleshoot the issue.  Also, I plan on making an option later to allow you to choose which controller to use.

**Comment by ChristianBliss** [comment]
@elishacloud that log is a godsend, thank so much!!!

Turns out that the device SH2 uses is a vJoy Device, even though I disabled those to begin with... Anyway, I got completely rid of the software (won't be using it for now) and BINGO! No more vJoy and I can use my controllers once again.

Thanks everybody for your time and help!! You are so kind to be helping the community through your work and troubleshooting.
